src/AngleStrategyVoterProxy.sol

```diff
@@ -0,0 +1,181 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity ^0.8.12;
+pragma experimental ABIEncoderV2;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {Address} from "@openzeppelin/contracts/utils/Address.sol";
+
+import {YearnAngleVoter} from "./YearnAngleVoter.sol";
+
+import "./interfaces/curve/ICurve.sol";
+import "./interfaces/Angle/IStableMaster.sol";
+import "./interfaces/Angle/IAngleGauge.sol";
+import "./interfaces/Uniswap/IUniV2.sol";
+
+library SafeVoter {
+    function safeExecute(
+        YearnAngleVoter voter,
+        address to,
+        uint256 value,
+        bytes memory data
+    ) internal {
+        (bool success, ) = voter.execute(to, value, data);
+        require(success);
+    }
+}
+
+contract AngleStrategyVoterProxy {
+    using SafeVoter for YearnAngleVoter;
+    using SafeERC20 for IERC20;
+    using Address for address;
+
+    YearnAngleVoter public yearnAngleVoter;
+    address public constant angleToken = address(0x31429d1856aD1377A8A0079410B297e1a9e214c2);
+
+    // gauge => strategies
+    mapping(address => address) public strategies;
+    mapping(address => bool) public voters;
+    address public governance;
+
+    constructor(address _voter) public {
+        governance = address(0xFEB4acf3df3cDEA7399794D0869ef76A6EfAff52);
+        yearnAngleVoter = YearnAngleVoter(_voter);
+    }
+
+    function setGovernance(address _governance) external {
+        require(msg.sender == governance, "!governance");
+        governance = _governance;
+    }
+
+    function approveStrategy(address _gauge, address _strategy) external {
+        require(msg.sender == governance, "!governance");
+        strategies[_gauge] = _strategy;
+    }
+
+    function revokeStrategy(address _gauge) external {
+        require(msg.sender == governance, "!governance");
+        strategies[_gauge] = address(0);
+    }
+
+    function approveVoter(address _voter) external {
+        require(msg.sender == governance, "!governance");
+        voters[_voter] = true;
+    }
+
+    function revokeVoter(address _voter) external {
+        require(msg.sender == governance, "!governance");
+        voters[_voter] = false;
+    }
+
+    function lock(uint256 amount, uint256 unlockTime) external {
```

Comment:
Do you see any harm adjusting this function to have zero arguments? My thought is that we always max lock the full balance. There is no access control on this function, what if someone locks for less than max? Do we care?

Reply:
I included a msg.sender check to it too to check for governance, I normally prefer to leave things as flexible as possible just in case. Wdyt?